metadata-ingestion/src/datahub/ingestion/source/qlik_sense/qlik_sense.py

```diff
@@ -0,0 +1,566 @@
+            }
+        )
+        return MetadataChangeProposalWrapper(
+            entityUrn=dataset_urn, aspect=dataset_properties
+        ).as_workunit()
+
+    def _get_app_table_identifier(self, table: QlikTable) -> Optional[str]:
+        if table.tableQri:
+            return table.tableQri.lower()
+        return None
+
+    def _gen_app_tables_workunit(
+        self, tables: List[QlikTable], app_id: str
+    ) -> Iterable[MetadataWorkUnit]:
+        for table in tables:
+            table_identifier = self._get_app_table_identifier(table)
+            if not table_identifier:
+                continue
+            dataset_urn = self._gen_qlik_dataset_urn(table_identifier)
+
+            yield self._gen_entity_status_aspect(dataset_urn)
+
+            yield self._gen_schema_metadata(table_identifier, table.datasetSchema)
+
+            yield self._gen_app_table_properties(dataset_urn, table)
+
+            yield from add_entity_to_container(
+                container_key=self._gen_app_key(app_id),
+                entity_type="dataset",
+                entity_urn=dataset_urn,
+            )
+
+            dpi_aspect = self._gen_dataplatform_instance_aspect(dataset_urn)
+            if dpi_aspect:
+                yield dpi_aspect
+
+            yield MetadataChangeProposalWrapper(
+                entityUrn=dataset_urn,
+                aspect=SubTypes(typeNames=[DatasetSubTypes.QLIK_DATASET]),
+            ).as_workunit()
+
+            upstream_lineage_workunit = self._gen_app_table_upstream_lineage(
+                dataset_urn, table
+            )
+            if upstream_lineage_workunit:
+                yield upstream_lineage_workunit
+
+    def _gen_app_workunit(self, app: App) -> Iterable[MetadataWorkUnit]:
+        """
+        Map Qlik App to Datahub container
+        """
+        owner_username = self.qlik_api.get_user_name(app.ownerId)
+        yield from gen_containers(
+            container_key=self._gen_app_key(app.id),
+            name=app.qTitle,
+            description=app.description,
+            sub_types=[BIContainerSubTypes.QLIK_APP],
+            parent_container_key=self._gen_space_key(app.spaceId),
+            extra_properties={Constant.QRI: app.qri, Constant.USAGE: app.qUsage},
+            owner_urn=builder.make_user_urn(owner_username)
+            if self.config.ingest_owner and owner_username
+            else None,
+            external_url=f"https://{self.config.tenant_hostname}/sense/app/{app.id}/overview",
+            created=int(app.createdAt.timestamp() * 1000),
+            last_modified=int(app.updatedAt.timestamp() * 1000),
+        )
+        yield from self._gen_sheets_workunit(app.sheets, app.id)
+        yield from self._gen_app_tables_workunit(app.tables, app.id)
+
+    def _gen_qlik_dataset_urn(self, dataset_identifier: str) -> str:
+        return builder.make_dataset_urn_with_platform_instance(
+            name=dataset_identifier,
```

Comment:
Is this the intended format? 

<img width="1108" alt="image" src="https://github.com/datahub-project/datahub/assets/12566801/d064cd47-a88b-4b89-9e6b-522c3840c5f7">

Reply:
On my side I don't see any indentation error.